components/core/src/clp/clo/clo.cpp

```diff
@@ -93,7 +93,9 @@ static SearchFilesResult search_files(
 
     // Run query on each file
     for (; file_metadata_ix.has_next(); file_metadata_ix.next()) {
-        if (segments_to_search.count(file_metadata_ix.get_segment_id()) == 0) {
+        if (query.contains_sub_queries()
```

Comment:
Do you not want to use clg's logic instead? Namely:

* if there are no subqueries:
  * search all files
* else if there are subqueries:
  * for each segment matching a subquery:
    * search each file within the segment
    
I think the logic above should be more efficient by avoiding some branches.

Reply:
Because we want to order files by timestamps, if we follow clg's logic to search each file within a segment, we cannot guarantee this order.

Reply:
Which branches do you think we can avoid?

Reply:
Ah, right, forgot about the timestamp sorting. We can still have separate cases for no-subqueries vs subqueries, right? That way we don't repeat the no-subqueries check for every file that we search.